tests/std/tests/GH_000625_vector_bool_optimization/test.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <assert.h>
+#include <vector>
+
+using namespace std;
+
+constexpr int blockSize = 32;
+static_assert(blockSize == _VBITS, "Invalid block size");
+
+void test_fill_helper(const size_t length) {
+    // No offset
+    {
+        vector<bool> result_true(length, true);
+        result_true.resize(length + 3, false);
+        vector<bool> dest_true(length + 3, false);
+        fill(dest_true.cbegin(), prev(dest_true.cend(), 3), true);
+        assert(dest_true == result_true);
+
+        vector<bool> result_false(length, false);
+        result_false.resize(length + 3, true);
+        vector<bool> dest_false(length + 3, true);
+        fill(dest_false.cbegin(), prev(dest_false.cend(), 3), false);

```

Comment:
`vector<bool>` is random-access, so stylistically I'd recommend just `iter - 3` instead of `prev(iter, 3)` throughout this test. If you really want to use `prev` that's fine though.

Reply:
Hm, working on ranges I am getting paranoid.